Source/Meadow.Contracts/ISleepAwarePeripheral.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace Meadow
+{
+    public interface ISleepAwarePeripheral
+    {
+        public Task BeforeSleep(CancellationToken cancellationToken);
```

Comment:
Love this - I personally lean towards `OnSleep` and `OnAwake` since that's all that's important to the peripheral.

But I also think the team in general prefers the prepositions so I may be out voted :)

Reply:
Also .... *cough* XML comments *cough* ;)